be/src/olap/rowset/segment_v2/inverted_index_compound_directory.cpp

```diff
@@ -392,10 +393,19 @@ DorisCompoundDirectory::FSIndexOutput::~FSIndexOutput() {
     if (writer) {
         try {
             FSIndexOutput::close();
+            DBUG_EXECUTE_IF(
+                    "DorisCompoundDirectory::FSIndexOutput._throw_clucene_error_in_fsindexoutput_"
+                    "destructor",
+                    {
+                        _CLTHROWA(CL_ERR_IO,
+                                  "debug point: test throw error in fsindexoutput destructor");
+                    })
         } catch (CLuceneError& err) {
             //ignore errors...
+            writer.reset(nullptr);
```

Comment:
If writer is a member of the Class. Then rename it to _writer OR writer_.
And why do you need reset here? since the unique ptr will decontructed automatically.

Reply:
Done